fetch.js

```diff
@@ -105,9 +105,34 @@
     this.bodyUsed = false
 
     if (blobSupport) {
+      this._initBody = function(body) {
+        this._bodyInit = body
+        if (typeof body === 'string') {
+          this._bodyText = body
+        } else if (Blob.prototype.isPrototypeOf(body)) {
+          this._bodyBlob = body
+        } else if (FormData.prototype.isPrototypeOf(body)) {
+          this._bodyFormData = body
+        } else if (!body) {
+          this._bodyText = ''
+        } else {
+          throw new Error('unsupported BodyInit type')
```

Comment:
That is because IDL handles it. And indeed, if there's a primitive value present that is what IDL defaults to.